### PR TITLE
chore: improve git url normalization

### PR DESCRIPTION
**Pull request by i4ki**

### CODE REVIEW

File: cloud/git_test.go [hunk 1]
```diff
@@ -66,14 +66,14 @@ func TestNormalizeGitURL(t *testing.T) {
 			normalized: "something else",
 		},
 		{
-			name:       "unrecognized ssh url",
+			name:       "unrecognized ssh url - THIS IS A BUG IN THE go-gh library",
```
Comment (i4ki) [inline cloud/git_test.go:69]: @mariux this is a bug that should never happen in the real world (I guess). Only if someone place an incorrect git URL in the remote configuration. 
The problem is that `git://` scheme does not support port numbers in the URL, in then the go-gh library parses it incorrectly. In order to use port number, the user must provide the `ssh://` protocol (or the `ssh+git://` protocol).

**Review by lennyterra — DISMISSED** [review]
LGTM :+1:

File: cloud/git_test.go (hunk 1 above)
Comment (mariux) [reply cloud/git_test.go:69]: hmmm can the go-gh library parse gitlab and co?

**Review by mariux — CHANGES_REQUESTED** [review]
does this break all non GH urls?

File: cloud/git_test.go (hunk 1 above)
Comment (i4ki) [reply cloud/git_test.go:69]: yes, just added tests covering that! Thanks

**Pull request merged**